bots.py

```diff
@@ -16,10 +16,22 @@ def run(room, bot, text, params, user=None):
     reply = requests.post(endpoint_url, json=data)
 
     if reply.status_code == requests.codes.ok:
+      safe_message = {
+        'author': name,
+      }
+
       message = reply.json()
-      if 'room' not in message:
-        message['room'] = room
-      return message
+      if isinstance(message, dict) and 'text' in message and isinstance(message['text'], str):
+        safe_message['text'] = message['text']
+      else:
+        return None
+
+      if 'room' in message and isinstance(message['room'], str):
+        safe_message['room'] = message['room']
+      else:
+        safe_message['room'] = room
+
+      return safe_message
```

Comment:
@kennib what do you think of these changes? It makes the handling of bot responses more robust and simpler. It also removes the ability of a bot to set a custom author in the response (not sure if this would be wanted).